awx/main/migrations/_galaxy.py

```diff
@@ -0,0 +1,85 @@
+    Setting = apps.get_model('conf', 'Setting')
+
+    galaxy_type = CredentialType.objects.get(kind='galaxy')
+    private_galaxy_url = Setting.objects.filter(key='PRIMARY_GALAXY_URL').first()
+
+    # by default, prior versions of AWX/Tower automatically pulled content
+    # from galaxy.ansible.com
+    public_galaxy_enabled = True
+    public_galaxy_setting = Setting.objects.filter(key='PUBLIC_GALAXY_ENABLED').first()
+    if public_galaxy_setting and public_galaxy_setting.value is False:
+        # ...UNLESS this behavior was explicitly disabled via this setting
+        public_galaxy_enabled = False
+
+    for org in Organization.objects.all():
+        if private_galaxy_url and private_galaxy_url.value:
+            # If a setting exists for a private Galaxy URL, make a credential for it
+            username = Setting.objects.filter(key='PRIMARY_GALAXY_USERNAME').first()
+            password = Setting.objects.filter(key='PRIMARY_GALAXY_PASSWORD').first()
+            if (username and username.value) or (password and password.value):
+                logger.error(
+                    f'Specifying HTTP basic auth for the Ansible Galaxy API '
+                    f'({private_galaxy_url.value}) is no longer supported. '
+                    'Please provide an API token instead after your upgrade '
+                    'has completed',
+                )
+            inputs = {
+                'url': private_galaxy_url.value
+            }
+            token = Setting.objects.filter(key='PRIMARY_GALAXY_TOKEN').first()
+            if token and token.value:
+                inputs['token'] = decrypt_field(token, 'value')
+            auth_url = Setting.objects.filter(key='PRIMARY_GALAXY_AUTH_URL').first()
+            if auth_url and auth_url.value:
+                inputs['auth_url'] = auth_url.value
+            name = f'Private Galaxy ({private_galaxy_url.value})'
+            if 'cloud.redhat.com' in inputs['url']:
+                name = f'Ansible Automation Hub ({private_galaxy_url.value})'
+            cred = Credential(
+                created=now(),
+                modified=now(),
+                name=name,
+                organization=org,
+                credential_type=galaxy_type,
+                inputs=inputs
+            )
+            cred.save()
+            if token and token.value:
+                # encrypt based on the primary key from the prior save
+                cred.inputs['token'] = encrypt_field(cred, 'token')
+                cred.save()
+            org.galaxy_credentials.add(cred)
+        if public_galaxy_enabled:
+            # If public Galaxy was enabled, make a credential for it
+            cred = Credential(
+                created=now(),
+                modified=now(),
+                name='Ansible Galaxy',
+                organization=org,
+                credential_type=galaxy_type,
+                inputs = {
+                    'url': 'https://galaxy.ansible.com/'
```

Comment:
I'm a little confused by this in particular. What happens if you create a new organization after this migration has finished? What credentials does it have associated, and how does it go about doing Galaxy commands before the user has associated any credentials?

Also, why would each organization get a _new_ public galaxy "credential"? Wouldn't all orgs just share the same credential? It's 1 URL.

Reply:
Also, shouldn't the public Galaxy credential be visible to all users? In this implementation, it seems like it would only be visible to superusers.

Reply:
> I'm a little confused by this in particular. What happens if you create a new organization after this migration has finished? What credentials does it have associated...

None.  We're changing the default behavior when you create new Organizations to *not* pull unofficial community content (cc @wenottingham in case I've misunderstood this).

Reply:
> Also, why would each organization get a new public galaxy "credential"? Wouldn't all orgs just share the same credential? It's 1 URL.

Okay, that's probably a better way to do this.  Who's allowed to edit that one global credential, though?

Reply:
How would each org be able to use this credential without weird backend logic on org creation?

Reply:
> How would each org be able to use this credential without weird backend logic on org creation?

Yea, I don't really want to go down that road.

Reply:
@AlanCoding @wenottingham @rebeccahhh @chrismeyersfsu 

Would it make sense to enforce something like "Galaxy Credentials must be assigned to one organization." ?  It would be fairly simple to do, and would prevent a number of edge cases.  I can't think of any reason why you'd want to create one of these as a personal credential - the entire point is to associate it with an Org.

Reply:
Another solution I think I like better (which I've pushed to this PR) is to just introduce the concept of `Credential.managed_by_tower`, just like we did with `CredentialType.managed_by_tower`, and to have one global, non-editable Credential that represents galaxy.ansible.com

Reply:
> What happens if you create a new organization after this migration has finished? What credentials does it have associated, and how does it go about doing Galaxy commands before the user has associated any credentials?

Could all new organizations start with a default credential in their Galaxy credential list? The default credential could point to `galaxy.ansible.com`.

Reply:
Alternatively (& I think I might like this might be preferable to what I mentioned above):

We might be able to address this by making sure we show a good & clear error message / warning if the user trys to run a JT that fetches content without having configured a galaxy credential. There's a variety of places we could surface such an error message (e.g job details page).

Reply:
> Could all new organizations start with a default credential in their Galaxy credential list? The default credential could point to galaxy.ansible.com.

What I've discussed with @wenottingham @AlanCoding and @mabashian is just this - we're going to make it so that the Org Creation form pre-fills with "Ansible Galaxy".